File: Sprint7/D/C++/main.cpp
// https://contest.yandex.ru/contest/25596/problems/D/
#include <iostream>
#include <vector>

const int MOD = 1e9 + 7;

int main() {
    int n;
    std::cin >> n;

    if (n == 0) {
        std::cout << "0" << std::endl;
        return 0;
    }

    std::vector<int> fibonacci(n + 1);
    fibonacci[0] = 1;
    fibonacci[1] = 1;

    for (int i = 2; i <= n; i++) {
        fibonacci[i] = (fibonacci[i - 1] + fibonacci[i - 2]) % MOD;
    }

    std::cout << fibonacci[n] << std::endl;

    return 0;
}

File: Sprint7/D/Rust/src/main.rs
// https://contest.yandex.ru/contest/25596/problems/D/

fn main() {
    let mut lines = std::io::stdin().lines();
    let n = lines.next().unwrap().unwrap().parse::<usize>().unwrap();

    if n == 0 {
        println!("0");
        return;
    }

    let mut fibonacci = Vec::with_capacity(n + 1);
    fibonacci.push(1i64);
    fibonacci.push(1i64);

    for i in 2..n + 1 {
        fibonacci.push((fibonacci.get(i - 1).unwrap() + fibonacci.get(i - 2).unwrap()) % i64::MAX);
    }

    println!("{}", fibonacci[n]);
}
